ams/optimizer/src/main/java/com/netease/arctic/optimizer/FlinkOptimizerContainer.java

```diff
@@ -72,15 +82,105 @@ protected Map<String, String> doScaleOut(String startUpArgs) {
 
   @Override
   protected String buildOptimizerStartupArgsString(Resource resource) {
-    String taskManagerMemory = PropertyUtil.checkAndGetProperty(resource.getProperties(),
-        TASK_MANAGER_MEMORY_PROPERTY);
-    String jobManagerMemory = PropertyUtil.checkAndGetProperty(resource.getProperties(), JOB_MANAGER_MEMORY_PROPERTY);
+    Map<String, String> properties = resource.getProperties();
+    Map<String, String> flinkConfig = loadFlinkConfig();
+    Preconditions.checkState(properties != null && !flinkConfig.isEmpty(),
+        "resource properties is null or load flink-conf.yaml failed");
+    Long jobManagerMemory = getMemorySizeValue(properties, flinkConfig, JOB_MANAGER_MEMORY_PROPERTY,
+        JOB_MANAGER_TOTAL_PROCESS_MEMORY);
+    Long taskManagerMemory = getMemorySizeValue(properties, flinkConfig, TASK_MANAGER_MEMORY_PROPERTY,
+        TASK_MANAGER_TOTAL_PROCESS_MEMORY);
     String jobPath = getAMSHome() + "/plugin/optimize/OptimizeJob.jar";
-    long memory = Long.parseLong(jobManagerMemory) + Long.parseLong(taskManagerMemory) * resource.getThreadCount();
-    return String.format("%s/bin/flink run -m yarn-cluster -ytm %s -yjm %s -c %s %s -m %s %s",
-        getFlinkHome(), taskManagerMemory, jobManagerMemory,
-        FlinkOptimizer.class.getName(), jobPath, memory,
-        super.buildOptimizerStartupArgsString(resource));
+    long memory = jobManagerMemory + taskManagerMemory * resource.getThreadCount();
+    return String.format("%s/bin/flink run -m yarn-cluster -ytm %s -yjm %s %s -c %s %s -m %s %s",
```

Comment:
Why don't we directly reuse Flink's parameter configuration for memory-related parameters? Why do we need to recalculate them?

Reply:
Firstly, the purpose of calculating memory in this area is to display resource usage on the front-end.

Secondly, due to the possibility of users using old parameters in history:` jobmanager.memory`，`taskmanager.memory`, in order to be compatible these users did not directly use the flink parameter to calculate the memory situation.

This is also a point that needs to be discussed, whether it is necessary to be compatible with the parameters configured by old users: `jobmanager.memory`、`taskmanager.memory`